.devcontainer/Dockerfile

```diff
@@ -0,0 +1,80 @@
+# Build stage with Spack pre-installed and ready to be used
+FROM spack/ubuntu-jammy as builder
+
+# Install OS packages needed to build the software
+RUN apt-get -yqq update && apt-get -yqq upgrade \
+ && apt-get -yqq install autoconf \
+ && rm -rf /var/lib/apt/lists/*
+
+# What we want to install and how we want to install it
+# is specified in a manifest file (spack.yaml)
+RUN mkdir /opt/spack-environment \
+&&  (echo spack: \
+&&   echo '  specs:' \
+&&   echo '  - exago@1.6.0+ipopt+python+mpi' \
+&&   echo '  - petsc~fortran~hdf5~hypre+metis' \
+&&   echo '  - ipopt@3.12.10+coinhsl~mumps+metis' \
+&&   echo '  - coinhsl@2019.05.21+blas' \
+&&   echo '  - py-jupyterlab' \
```

Comment:
2336.9s was how long it took me to build on my laptop, which installed 171 packages.  

cc @paulrigor and @Jayapreethi this is the sort of use case we need to support and speed up for users if possible

Reply:
@eugeneswalker maybe E4S can help here?

We can't use CoinHSL, but it would at least be nice to cache some of these packages to speed up people's devcontainer builds when they want to try things out for themselves. 

SInce E4S support ends after ECP, we need a sustainable solution that ExaGO can adopt independently after project ends at low cost.

cc @pelesh as we are almost ready with viable CoinHSL alternative, which would allow us to provide true rolling ExaGO releases, squashing 171 packages into needing <10...

Reply:
> @eugeneswalker maybe E4S can help here?

Yes, we can help by putting as many of these packages into a cache as possible. This container is intended to run on intel or arm macOS, or both and more?

Reply:
With dev-containers and container runtimes, I was under the impression the we would just need ubuntu x86_64 binaries, and then those would effectively be cross platform?

That being said, we also want to support users running CPU only builds on their laptops with all the architectures that you listed, so if MacOS is within reach I would love that too.

#52 is working towards this in that at least having GitHub actions for Mac would be good, but ideally supporting this fully would be what we do